tsconfig.packages.build.jsonc

```diff
@@ -0,0 +1,16 @@
+{
+  "extends": "./tsconfig.packages.json",
+  "compilerOptions": {
+    "declaration": true,
+    "declarationMap": true,
+    "emitDeclarationOnly": true,
+    "inlineSources": true,
+    "sourceMap": true,
+    // Prevent automatic inclusion of @types packages by using an empty array
+    // This gives us more control over type definitions and helps avoid conflicts
+    // Types must be explicitly imported in files where they are needed
+    // or included in extending tsconfig files for specific parts of the project
```

Comment:
JSON typically doesn't allow comments. Updating `tsconfig.packages.build.json` to `jsonc` to allow for critical comments about our typescript configuration